### PR TITLE
test: sanity check storage setup in estimator

### PR DESCRIPTION
**Pull request by jakmeier**
Add a simple test in the estimator that runs 3 cheap estimations,
including the state dump setup.

Also add a (currently ignored) check that flat storage works.

### CODE REVIEW

File: runtime/runtime-params-estimator/Cargo.toml [hunk 1]
```diff
@@ -69,9 +69,13 @@ nightly_protocol = [
   "near-primitives/nightly_protocol",
   "near-test-contracts/nightly",
   "protocol_feature_nep366_delegate_action",
+  "protocol_feature_flat_state",
 ]
 protocol_feature_nep366_delegate_action = [
   "nearcore/protocol_feature_nep366_delegate_action",
 ]
+protocol_feature_flat_state = [
+  "nearcore/protocol_feature_nep366_delegate_action",
```
Comment (pugachAG) [inline runtime/runtime-params-estimator/Cargo.toml:78]: `protocol_feature_nep366_delegate_action` -> `protocol_feature_flat_state`

File: runtime/runtime-params-estimator/Cargo.toml [hunk 2]
```diff
@@ -69,9 +69,13 @@ nightly_protocol = [
   "near-primitives/nightly_protocol",
   "near-test-contracts/nightly",
   "protocol_feature_nep366_delegate_action",
+  "protocol_feature_flat_state",
```
Comment (pugachAG) [inline runtime/runtime-params-estimator/Cargo.toml:72]: I believe this should be a part of `nightly` to be consistent with `nearcore/Cargo.toml`

File: runtime/runtime-params-estimator/Cargo.toml (hunk 1 above)
Comment (jakmeier) [reply runtime/runtime-params-estimator/Cargo.toml]: oops, thanks!

File: runtime/runtime-params-estimator/Cargo.toml (hunk 2 above)
Comment (jakmeier) [reply runtime/runtime-params-estimator/Cargo.toml]: Hm, makes sense. (Although, I must admit I'm never sure what the difference between "nightly" and "nightly_protocol" is anyway.)

File: runtime/runtime-params-estimator/src/utils.rs [hunk 3]
```diff
@@ -203,6 +203,25 @@ pub(crate) fn fn_cost_with_setup(
 
         let (gas_cost, ext_costs) =
             aggregate_per_block_measurements(block_size, measurements, Some(overhead));
+
+        // flat storage check: we only expect TTN costs for writes
+        #[cfg(feature = "protocol_feature_flat_state")]
+        // TODO(#7327): This assertion is ignored, we know flat storage doesn't
```
Comment (pugachAG) [inline runtime/runtime-params-estimator/src/utils.rs:209]: Actually I'm working on enabling flat storage for runtime-params-estimator as part of #8322, so you might want to update `TODO` here :)

**Review by pugachAG — APPROVED** [review]
LGTM!

File: runtime/runtime-params-estimator/src/utils.rs (hunk 3 above)
Comment (jakmeier) [reply runtime/runtime-params-estimator/src/utils.rs]: done!

**Review by Longarithm — APPROVED** [review]

**Pull request merged**